MobiusCore/Source/EventProcessor.swift

```diff
@@ -21,14 +21,14 @@ import Foundation
 
 /// Internal class that manages the atomic state updates and notifications of model changes when processing of events via
 /// the Update function.
-class EventProcessor<Types: LoopTypes>: Disposable, CustomDebugStringConvertible {
-    let update: Update<Types>
-    let publisher: ConnectablePublisher<Next<Types.Model, Types.Effect>>
+class EventProcessor<Model, Event, Effect>: Disposable, CustomDebugStringConvertible {
+    let update: Update<Model, Event, Effect>
```

Comment:
Here we can see that instead of doing `Types: LoopTypes` we have to do `Model, Event, Effect`, but perhaps that isn't too different. I believe it is more typical for library level code to have to worry about the generics, whereas client code can usually use type inference to not have to worry about it.

Also worth noting that we get to remove some `Types.` noise when needing to refer to the generics inside `Types`.